Assets/Unium/Core/UniumSocketMessage.cs

```diff
@@ -17,23 +17,22 @@ public class Message
 
             public class Repeat
             {
-                public int      skip    = 0;
-                public int      samples = int.MaxValue;
-                public float    freq    = 1.0f;
-                public bool     cache   = false;
+                public int   skip    = 0;
+                public int   samples = int.MaxValue;
+                public float freq    = 1.0f;
+                public bool  cache   = false;
             }
 
-            public string   id;             // arbitary message identifier
-            public string   q;              // full query (uri + optional query string parameters)
-            private int     queryStringIndex           { get { return q.IndexOf('?'); } }
-            public string   uri                        { get { return queryStringIndex >= 0 ? q.Substring(0, queryStringIndex) : q; } }
-            public string   queryParameters            { get { return queryStringIndex >= 0 ? q.Substring(queryStringIndex + 1) : string.Empty; } }
+            public string   id;                         // arbitary message identifier
+            public string   q;                          // full query ( <Path>[?<Parameters>] )
             public Repeat   repeat;
 
+            public string   Path                        { get { var i = q.IndexOf( '?' ); return i >= 0 ? q.Substring( 0, i ) : q; } }

```

Comment:
So thinking about it, `q` isn't a great name if we are treating it as a standard URI.

However if we are, then the component parts would be `path` and `query` if we stick to the [conventional nomenclature](https://en.wikipedia.org/wiki/URL#Syntax). Given throughout unium `query` is used in references to GQL that would be confusing, so Path and Parameters seem better for me. URI meaning `path` is a bit confusing.

CapWords as they are accessors.